src/test/scala/com/signalcollect/triplerush/sparql/Sparql11SelectSyntaxSpec.scala

```diff
@@ -0,0 +1,139 @@
+    tr.load(tmpDir.toString + File.separator + manifestFile)
+    tr.awaitIdle
+    tr.prepareExecution
+    //Retrieve sub-manifests
+    val subManifestQuery =
+      """|PREFIX rdf:    <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+        |PREFIX list:   <http://jena.hpl.hp.com/ARQ/list#>
+        |PREFIX mf:     <http://www.w3.org/2001/sw/DataAccess/tests/test-manifest#>
+        |PREFIX qt:     <http://www.w3.org/2001/sw/DataAccess/tests/test-query#>
+        |SELECT DISTINCT ?subManifest
+        |WHERE {
+        |  <http://www.w3.org/TR/sparql11-query/> mf:conformanceRequirement ?list .
+        |  ?list list:member ?subManifest .
+        |}
+        | """.stripMargin
+    val subManifestsResultSet = Sparql(subManifestQuery)
+    val subManifests = subManifestsResultSet.map(f => f.get("subManifest").toString).toList
+    //Load sub-manifests.
+    subManifests.map {
+      subManifest =>
+        val subManifestFile = subManifest.replace("file://", "")
+        tr.load(subManifestFile)
+        tr.awaitIdle
+    }
+    tr.prepareExecution
+    //Retrieve location of query to run and type(whether it could parse or not).
+    val query =
+      """
+        |PREFIX mf:  <http://www.w3.org/2001/sw/DataAccess/tests/test-manifest#>
+        |PREFIX qt:  <http://www.w3.org/2001/sw/DataAccess/tests/test-query#>
+        |PREFIX dawgt: <http://www.w3.org/2001/sw/DataAccess/tests/test-dawg#>
+        |PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+        |SELECT ?queryToRun ?type
+        |WHERE { [] rdf:first ?testURI.
+        |        ?testURI a ?type ;
+        |        mf:action ?queryToRun ;
+        |        dawgt:approval dawgt:Approved .
+        |        FILTER(?type IN (mf:PositiveSyntaxTest11, mf:NegativeSyntaxTest11, mf:PositiveUpdateSyntaxTest11, mf:NegativeUpdateSyntaxTest11))
+        |}
+      """.stripMargin
+
+    val results = Sparql(query)
+
+    case class Test(queryToRun: String, positive: Boolean)
+
+    val testsToRun = results.map(test => {
+      val queryToRun = test.get("queryToRun").toString
+      val typeOfTest = test.get("type").toString
+      val positiveTest = typeOfTest.equals("http://www.w3.org/2001/sw/DataAccess/tests/test-manifest#PositiveSyntaxTest11") ||
+        typeOfTest.equals("http://www.w3.org/2001/sw/DataAccess/tests/test-manifest#PositiveUpdateSyntaxTest11")
+      Test(queryToRun, positiveTest)
+    }).toList
+
+    var expectedNumOfPositiveTests = 0
+    val expectedNumOfNegativeTests = testsToRun.count(p => !p.positive)
+    var actualNumOfPositivePassed = 0
+    var actualNumOfNegativePassed = 0
+    testsToRun.map(test => {
+      val query = scala.io.Source.fromFile(test.queryToRun.replace("file://", "")).mkString
+      if (test.positive) {
+        try {
+          val queryFactoryQuery = QueryFactory.create(query)
+          if (queryFactoryQuery.isSelectType && !query.contains("SERVICE")) {
+            expectedNumOfPositiveTests += 1
+            Sparql(query)
```

Comment:
definitely. still this gives us a room if at all we write our own parser(anytime). I will add the results test as well.